doc/src/manual/style-guide.md

```diff
@@ -164,6 +164,62 @@ uses (e.g. `a[i]::Int`) than to try to pack many alternatives into one type.
 If a function name requires multiple words, consider whether it might represent more than one
 concept and might be better split into pieces.
 
+## Write functions that use similar argument precedence as Julia's `base/`
+
+As a general rule, the Base library uses the following order of arguments to functions,
+as applicable:
+
+1. **Function argument**.
+   This permits the use of `do` blocks for passing multiline anonymous functions.
+
+2. **I/O stream**.
+   Specifying the `IO` object first permits passing the function to functions such as
+   [`sprint`](@ref), e.g. `sprint(show, x)`.
+
+3. **Input being mutated**.
+   In [`fill!(x, v)`](@ref fill!), `x` is the object being mutated, and it appears before
+   the value to be inserted into `x`.
+
+4. **Type**.
+   Passing a type typically means that the output will have the given type.
+   In [`read(io, String)`](@ref read), the `IO` argument appears before the type.
+
+5. **Input not being mutated**.
+   In `fill!(x, v)`, `v` is *not* being mutated and it comes after `x`.
+
+6. **Key**.
+   For associative collections, this is the key of the key-value pair(s).
+   For other indexed collections, this is the index.
+
+7. **Value**.
+   For associative collections, this is the value of the key-value pair(s).
+   In cases like `fill!(x, v)`, this is `v`.
+
+8. **Everything else**.
+   Any other arguments.
+
+9. **Varargs**.
+   This refers to arguments that can be listed indefinitely at the end of a function call.
+   For example, in `Matrix{T}(uninitialized, dims)`, the dimensions can be given as a
+   `Tuple`, e.g. `Vector{T}(uninitialized, (1,2))`, or as `Varargs`, e.g.
```

Comment:
Could `@ref` `Tuple` and `Varargs`

Reply:
`Varargs` doesn't have docs but I'll do it for `Tuple`.

Reply:
It does:
```
help?> Vararg
search: Vararg

  Vararg{T,N}

  The last parameter of a tuple type Tuple can be the special type Vararg, which denotes any number of trailing elements. The type Vararg{T,N} corresponds to exactly N elements of type T. Vararg{T} corresponds
  to zero or more elements of type T. Vararg tuple types are used to represent the arguments accepted by varargs methods (see the section on Varargs Functions in the manual.)

  Examples
  ≡≡≡≡≡≡≡≡≡≡

  julia> mytupletype = Tuple{AbstractString,Vararg{Int}}
  Tuple{AbstractString,Vararg{Int64,N} where N}
  
  julia> isa(("1",), mytupletype)
  true
  
  julia> isa(("1",1), mytupletype)
  true
  
  julia> isa(("1",1,2), mytupletype)
  true
  
  julia> isa(("1",1,2,3.0), mytupletype)
  false
```

Reply:
Uh weird, I thought I tried that yesterday. Okay, I'll ref that too.

Reply:
Oh, because I misspelled it.